### PR TITLE
Added Rabbit Metadata values.

### LINKED ISSUE #1867: Only brown rabbits spawn
When spawning rabbits, only brown colored rabbits spawn, instead of rabbits in different colors and patterns.

### PR DESCRIPTION
**Pull request by bibo38**
Added the Rabbit Types and the MoreCarrotTicks value. Types are selected randomly on creation.
Fixes #1867

### CODE REVIEW

File: src/Mobs/Rabbit.cpp [hunk 1]
```diff
@@ -9,8 +9,27 @@
 
 
 
+static cFastRandom Random;
```
Comment (worktycho) [inline src/Mobs/Rabbit.cpp]: Are you sure this is safe? The constructors can be called from plugins, and cFastRandom is not thread safe.

File: src/WorldStorage/NBTChunkSerializer.cpp [hunk 2]
```diff
@@ -610,6 +610,13 @@ void cNBTChunkSerializer::AddMonsterEntity(cMonster * a_Monster)
 				m_Writer.AddByte("IsConverting", (((const cZombie *)a_Monster)->IsConverting() ? 1 : 0));
 				break;
 			}
+			case mtRabbit:
+			{
+				const cRabbit & Rabbit = *((const cRabbit *) a_Monster);
```
Comment (worktycho) [inline src/WorldStorage/NBTChunkSerializer.cpp]: reinterpert_cast?

**Comment by worktycho** [comment]
@madmaxoft Do you know what the problem with appveyor is? Its complaining about platform tools.

File: src/Mobs/Rabbit.cpp (hunk 1 above)
Comment (NiLSPACE) [inline src/Mobs/Rabbit.cpp]: cFastRandom isn't exported to the API.

Comment (worktycho) [inline src/Mobs/Rabbit.cpp]: No, I meant cRabbit's constructors. They can be called through cWorld::SpawnMob

Comment (NiLSPACE) [inline src/Mobs/Rabbit.cpp]: Oh, right, I didn't think of that.

File: src/WorldStorage/NBTChunkSerializer.cpp (hunk 2 above)
Comment (bibo38) [inline src/WorldStorage/NBTChunkSerializer.cpp]: I wanted to keep consistent with the surrounding code. None of them uses reinterpretet casts.

Comment (worktycho) [inline src/WorldStorage/NBTChunkSerializer.cpp]: Then change the surrounding code. We are moving over from c style casts, its just there are a lot of them to onvert.

**Comment by SphinxC0re** [comment]
Any updates on this?

**Comment by bibo38** [comment]
If everything is ok, I would squash the PR into a single commit. Then it should be able getting merged.

**Pull request merged**